code/game/objects/items/weapons/weaponry.dm

```diff
@@ -163,29 +163,77 @@ obj/item/weapon/wirerod/attackby(obj/item/I, mob/user, params)
 	w_class = WEIGHT_CLASS_HUGE
 	var/homerun_ready = 0
 	var/homerun_able = 0
+	var/swing_power = 0
+	var/at_bat = FALSE
 
 /obj/item/weapon/melee/baseball_bat/homerun
 	name = "home run bat"
 	desc = "This thing looks dangerous... Dangerously good at baseball, that is."
 	homerun_able = 1
 
 /obj/item/weapon/melee/baseball_bat/attack_self(mob/user)
+	wind_up(user)
+	..()
+
+/obj/item/weapon/melee/baseball_bat/proc/wind_up(mob/user)
+	if(at_bat)
+		return
+	if(swing_power == 3)
+		return
+	at_bat = TRUE
+	to_chat(user, "<span class='notice'>You begin gathering strength...</span>")
+	if(swing_power == 0  && at_bat)
+		if(do_after(user, 10, target = src) && at_bat)
+			swing_power = 1
+			user.playsound_local(null, 'sound/items/timer.ogg', 50, 0)
+	if(swing_power == 1  && at_bat)
+		if(do_after(user, 20, target = src) && at_bat)
+			swing_power = 2
+			user.playsound_local(null, 'sound/items/timer.ogg', 50, 0)
+	if(swing_power == 2  && at_bat)
+		if(do_after(user, 40, target = src) && at_bat)
+			swing_power = 3
+			user.playsound_local(null, 'sound/items/timer.ogg', 50, 0)
+	at_bat = FALSE
+	spawn(30)

```

Comment:
Don't use spawns; we have an `addtimer` system for this now.

Reply:
Not arguing against using the timers, but for the sake of information, wouldn't a timer be more expensive in this case as you'd be adding additional proc calls?

Reply:
`spawn` is pretty expensive, in and of itself; it's more than just a single proc call.